From b2de616aa714fe34edaf2ae1f0369dedb6496f64 Mon Sep 17 00:00:00 2001
From: Dongwon Kim <dongwon.kim@intel.com>
Date: Wed, 10 Apr 2024 09:34:00 -0700
Subject: [PATCH 8/8] modesetting: Empty the damage after updating front-buffer

Damage needs to be emptied after the update is done. Without it,
there is some leakage, which will affect the perf.

Upstream-Status: Submitted [https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/1486]

Signed-off-by: Dongwon Kim <dongwon.kim@intel.com>
---
 hw/xfree86/drivers/modesetting/driver.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/hw/xfree86/drivers/modesetting/driver.c b/hw/xfree86/drivers/modesetting/driver.c
index 6df62e38a..2a1f7b8c5 100644
--- a/hw/xfree86/drivers/modesetting/driver.c
+++ b/hw/xfree86/drivers/modesetting/driver.c
@@ -654,6 +654,8 @@ dispatch_dirty(ScreenPtr pScreen)
             return;
         }
     }
+
+    DamageEmpty(ms->damage);
 }
 
 static void
-- 
2.34.1

